### PR TITLE
Authorization for accountid and containerid in store server.

### PR DESCRIPTION
**Pull request by zzmao**
AmbryRequests handle this.

For get, do it after store.get() and check authorization in handleGetRequest.
For delete, do it after get the requested indexValue. 

May need one more unit test to test GET AuthorizationFailure case. 

Authorization_Failure(Store error code) is mapped to Blob_Not_Found(Server error code.)

### CODE REVIEW

File: ambry-store/src/main/java/com.github.ambry.store/IndexValue.java [hunk 1]
```diff
@@ -180,6 +180,12 @@ private IndexValue(long size, Offset offset, byte flags, long expiresAtMs, long
     version = PersistentIndex.CURRENT_VERSION;
   }
 
+  public boolean validateAuthorization(short accountId, short containerId) {
```
Comment (vgkholla) [inline ambry-store/src/main/java/com.github.ambry.store/IndexValue.java]: doesn't belong here - lets put it in a utils file

Comment (vgkholla) [reply ambry-store/src/main/java/com.github.ambry.store/IndexValue.java]: can be as simple as passing two account ids and containers ids and getting them matched if the first account id and first container id are not unknown. Just have to make sure we pass the account/container ID from the store as the "first" account and container id

File: ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java [hunk 2]
```diff
@@ -575,8 +579,27 @@ public void deleteErrorCasesTest() {
     verifyDeleteFailure(deletedKeys.iterator().next(), StoreErrorCodes.ID_Deleted);
     // ID that does not exist
     verifyDeleteFailure(getUniqueId(), StoreErrorCodes.ID_Not_Found);
+    // accoundId and containerId not match
+    // TODO
```
Comment (vgkholla) [inline ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java]: whats the "todo" here?

File: ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java [hunk 3]
```diff
@@ -1177,8 +1204,12 @@ private void verifyPutFailure(MockId idToPut, StoreErrorCodes expectedErrorCode)
    * @param expectedErrorCode the expected {@link StoreErrorCodes} for the failure.
    */
   private void verifyDeleteFailure(MockId idToDelete, StoreErrorCodes expectedErrorCode) {
-    MessageInfo info = new MessageInfo(idToDelete, DELETE_RECORD_SIZE, Utils.getRandomShort(TestUtils.RANDOM),
-        Utils.getRandomShort(TestUtils.RANDOM), System.currentTimeMillis());
+    short accoundId = accountIdMap.containsKey(idToDelete) == true ? accountIdMap.get(idToDelete)
```
Comment (vgkholla) [inline ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java]: didn't understand this - why do this?

File: ambry-store/src/main/java/com.github.ambry.store/IndexValue.java (hunk 1 above)
Comment (zzmao) [reply ambry-store/src/main/java/com.github.ambry.store/IndexValue.java]: sounds good.

File: ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java (hunk 2 above)
Comment (zzmao) [reply ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java]: ignore it. I forgot to remove.

File: ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java (hunk 3 above)
Comment (zzmao) [reply ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java]: The reason to create hashMap is we want to match accountId and containerId. Previously test code always use random.

However, some test gives unknown blobId to test ID_Not_Found behavior.
For example, verifyDeleteFailure(getUniqueId(), StoreErrorCodes.ID_Not_Found);
ID generated by getUniqueId() doesn't exist in HashMap. null returned by hashMap is not compatible with short, which required by MessageInfo().

**Comment by vgkholla** [comment]
@zzmao something went wrong with your merge I think - you have a lot of changes in this PR that aren't yours

**Comment by zzmao** [comment]
Right. Something weird after rebase.

**Pull request closed without merging**